functions/src/common/query-iterator.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2024 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Query,
+  QueryDocumentSnapshot,
+  QuerySnapshot,
+} from 'firebase-admin/firestore';
+
+export class QueryIterator implements AsyncIterator<QueryDocumentSnapshot> {
```

Comment:
Please add TSDoc to all public classes and methods. Example generated from this code using Gemini:

```ts
/**
 * An async iterator that iterates over documents in a Firestore query.
 * 
 * This iterator fetches documents in batches, using the provided `pageSize`.
 * It orders the documents by the specified `orderField`.
 *
 * @example
 * ```typescript
 * const query = firestore.collection('cities').where('population', '>', 1000000);
 * const iterator = new QueryIterator(query, 100, 'population');
 *
 * for await (const doc of iterator) {
 *   console.log(doc.id, doc.data());
 * }
 * ```
 */
```